app/grandchallenge/reader_studies/models.py

```diff
@@ -879,6 +977,27 @@ def statistics(self):
             "questions": questions,
         }
 
+    @cached_property
+    def values_for_interfaces(self):
+        cache_key = f"{self.slug}-{self.modified.timestamp()}"
+        cached = cache.get(cache_key)
+        if cached:
+            return cached
+        interfaces = self.display_sets.values_list(
+            "values__interface__slug", flat=True
+        )
+        values_for_interfaces = {}
+        for interface in interfaces:
+            values = ComponentInterfaceValue.objects.none()
+            for ds in self.display_sets.all():
+                values |= ds.values.filter(interface__slug=interface)
+            values_for_interfaces[interface] = values
+        cache.set(cache_key, values_for_interfaces)
```

Comment:
We could probably use a timeout of 0 here actually, since the cache key includes the modification timestamp. I'd say we can cache this until the timestamp changes and then maybe we should actively remove any existing cache keys for this reader study?

Reply:
Does the modified time change when any of the values of the display sets change?